### DISCUSSION TITLE
Flink写入Doris，使用Doris的Aggregate数据模型，出现数据重复问题

### DISCUSSION THREAD

**Discussion by weakdouqing** [apache/doris#24409] [A - General / Q&A]
Flink Version:1.14.0
Doris Version:1.2.0
flink-doris-connector-1.14_2.12
Doris数据模型：Aggregate

### 问题：
flink处理完数据写入Doirs中，由Doris最终根据key完成数据的聚合，但当flink任务运行过程中失败从检查点恢复的时候，doris这个时间段的数据发生重复写入

### flink写doris的sink相关配置：
public static SinkFunction<RowData> getDorisSink(String dbName, String tableName){
        properties.put("sink.properties.column_separator", ",");
        properties.put("sink.properties.escape_delimiters", "true");
        properties.put("strict_mode", "false");
        properties.put("max_filter_ratio", "1");
        properties.put("sink.label-prefix", "xxx");
        properties.put("sink.enable-2pc", "true");

        SinkFunction<RowData> dorisSink = DorisSink.sink(
                fields,
                logicalTypes,
                DorisReadOptions.builder().build(),
                DorisExecutionOptions.builder()
                        .setBatchSize(1000000)
                        .setMaxRetries(3)
                        .setBatchIntervalMs(15000L)
                        .setStreamLoadProp(properties)
                        .build(),
                DorisOptions.builder()
                        .setFenodes("xxx")
                        .setTableIdentifier(dbName+"."+tableName)
                        .setUsername("xxx")
                        .setPassword("xxx")
                        .build()
        );
        return dorisSink;
    }

### 希望：
在不改变Aggregate数据模型的基础上，我应该修改哪些配置来解决数据重复的问题

**Comment by yagagagaga** [comment]
What's your flink-doris-connector version?  Your Doris version is outdated and not a stable version. Can you upgrade to the latest version?

**Reply by weakdouqing** [reply]
flink-doris-connector version：1.0.3
那我该升级到哪个版本能够解决掉这个问题呢？

**Reply by ixzc** [reply]
You can find it in the link below.
https://github.com/apache/doris-flink-connector/releases

**Reply by ixzc** [reply]
and this : https://repository.apache.org/content/repositories/snapshots/org/apache/doris/

**Reply by weakdouqing** [reply]
感谢您的建议，我将会尝试升级来解决这个问题

**Comment by yagagagaga** [comment]
> flink-doris-connector version：1.0.3 那我该升级到哪个版本能够解决掉这个问题呢？

You can upgrade to Doris 1.2.7 and flink-doris-connector 1.4.0.

**Reply by weakdouqing** [reply]
感谢您的建议，我将会按照这个建议版本尝试升级来解决这个问题
